docs/explanation/lifecycle-events.md

```diff
@@ -0,0 +1,57 @@
+# Lifecycle Events
+
+Juju events allow progression of the charm in its lifecycle and encapsulates part of the execution
+context of a charm. Below is the list of observed events for wordpress-k8s charm with how the charm
+reacts to the event. For more information about the charm’s lifecycle in general, refer to the
+charm’s life [documentation](https://juju.is/docs/sdk/a-charms-life#heading--the-graph).
+
+### start
+
+This event marks the charm’s state as started. The charm’s running state must be persisted by the
+charm in its own state. See the documentation on the
+[start event](https://juju.is/docs/sdk/start-event).
+
+### uploads_storage_attached
+
+This event marks the charm’s storage availability. The name of the event derived from the name of
+the storage noted in the `metadata.yaml` configuration under "storage" key.
+`containers.wordpress.mounts.storage` and `storage.uploads` section. The storage filesystem maps to
+`/var/www/html/wp-content/uploads` directory of the WordPress application, which is used to store
+uploaded content from the WordPress user.
+
+### leader_elected
+
+This event is fired when juju elects a leader unit among the replica peers. Wordpress-k8s charm
+then responds by setting up secrets and sharing them with peers through peer relation databag if
+not already set.
+
+### config-changed
+
+WordPress-k8s charm reacts to any configuration change and runs reconciliation between the current
+state and the desired state. See the list of configurations in the reference documentation.
```

Comment:
Should this include a link to the configuration page?

Reply:
Added link to configurations tab! Thanks for the comment :)